framework/auth/core.py

```diff
@@ -843,6 +849,52 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def clean_email_verifications(self):
+        email_verifications = deepcopy(self.email_verifications)
+        for token in self.email_verifications:
+            try:
+                self.verify_token(token)
+            except (KeyError, ExpiredTokenError):
+                email_verifications.pop(token)
+        self.email_verifications = email_verifications
+        self.save()
+
+    def verified_email_get(self):
+        """Called at login to see if there are emails to add or users to merge.  Delete expired tokens.
+        methods: GET
+        """
+        self.clean_email_verifications()
+        verified_emails = []
+        email_verifications = deepcopy(self.email_verifications)
+        for token in email_verifications:
+            if self.email_verifications[token]['confirmed']:
+                try:
+                    user_merge = User.find_one(Q('emails', 'eq', self.email_verifications[token]['email'].lower()))
+                except NoResultsFound:
+                    user_merge = False
+
+                verified_emails.append({'address': self.email_verifications[token]['email'],
+                                        'token': token,
+                                        'confirmed': self.email_verifications[token]['confirmed'],
+                                        'user_merge': user_merge.email if user_merge else False})
+        return verified_emails
+
+    def verify_token(self, token):
```

Comment:
This seems to do almost the exact same thing as `User._get_unconfirmed_email_for_token`. It'd probably make sense to just rename that method to "unprivatize" it, i.e. remove the leading underscore, and just use that instead of reimplementing it here.